### PR TITLE
Rinas/add report to task force

### PR DESCRIPTION
**Pull request by rinashalibo**

### CODE REVIEW

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 1]
```diff
@@ -15,10 +15,28 @@
...
+    "BQ80_trim_ends": "X_SCORE >= 7.9 and X_EDIST <= 5 and X_INDEX > 12 and X_INDEX < (X_LENGTH - 12)",
+    "CSKP_trim_ends": "X_SCORE >= 10 and X_EDIST <= 5 and X_INDEX > 12 and X_INDEX < (X_LENGTH - 12)",
+    "BQ60_mixed_only": "X_SCORE >= 6 and X_EDIST <= 5 and is_mixed",
+    "BQ80_mixed_only": "X_SCORE >= 7.9 and X_EDIST <= 5 and is_mixed",
+    "CSKP_mixed_only": "X_SCORE >= 10 and X_EDIST <= 5 and is_mixed",
+    "BQ80_trim_ends_mixed_only": "X_SCORE >= 7.9 and X_EDIST <= 5 and X_INDEX > 12 \
+        and X_INDEX < (X_LENGTH - 12) and is_mixed",
+    "CSKP_trim_ends_mixed_only": "X_SCORE >= 10 and X_EDIST <= 5 and X_INDEX > 12 \
+        and X_INDEX < (X_LENGTH - 12) and is_mixed",
+}
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py:42]: "is_mixed" is obsolete

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 2]
```diff
@@ -15,10 +15,28 @@
...
+
 
 def create_data_for_report(
     xgb_classifier: xgb.XGBClassifier,
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py]: In the rest of the code the classifier is generalized to be an sklearn classifier and not necessarily xgboost, is there anything XGBoost specific here? If not please change to just "classifier"

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 3]
```diff
@@ -63,7 +82,7 @@ def create_data_for_report(
         predictions_df[f"XGB_prob_{label}"] = probs[:, label]
         predictions_df[f"XGB_qual_{label}"] = quals[:, label]
         predictions_df[f"XGB_prediction_{label}"] = predictions[:, label]
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py]: ML_QUAL, ML_prob etc.

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 4]
```diff
@@ -63,7 +82,7 @@ def create_data_for_report(
...
-    df = pd.concat([X, predictions_df.drop(["label"], axis="columns", inplace=False)], axis=1)
+    df = pd.concat([X, predictions_df], axis=1)
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py:89]: This is kind of an awkward way to do that. If you are just assigning columns this should do:

`df = X.assign(**{f"XGB_prob_{label}": probs[:, label] for label in labels}).assign(**{f"XGB_qual_{label}": quals[:, label] for label in labels}).assign(**{f"XGB_prediction_{label}": predictions[:, label] for label in labels})`

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 5]
```diff
@@ -84,24 +103,48 @@ def create_data_for_report(
...
+        -p {output_precision_recall_qual} \
+        -p {output_qual_density} \
+        -p {output_obsereved_qual_plot} \
+        -p {output_ML_qual_hist} \
+        -p {output_qual_per_feature} \
+        -p {output_bepcr_hists} \
+        -p {output_bepcr_fpr} \
+        -p {output_bepcr_recalls} \
+        -k python3",
         f"jupyter nbconvert {reportfile} --output {reporthtml} --to html --template classic --no-input",
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py:156]: The "--template classic" argument is deprecated, you can remove it

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 6]
```diff
@@ -140,9 +183,10 @@ def plot_ROC_curve(
 
     recall = {}
     precision = {}
+
     for xvar, querystr, namestr in (
         ("X_SCORE", "X_SCORE>-1", "X_SCORE"),
-        ("X_SCORE", "is_mixed == True & X_SCORE>-1", "X_SCORE_mixed"),
+        # ("X_SCORE", "is_mixed == True & X_SCORE>-1", "X_SCORE_mixed"),
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py]: Delete commented line

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 7]
```diff
@@ -140,9 +183,10 @@ def plot_ROC_curve(
 
     recall = {}
     precision = {}
+
     for xvar, querystr, namestr in (
         ("X_SCORE", "X_SCORE>-1", "X_SCORE"),
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py]: Please use the enumeration from featuremap_utils instead of the string "X_SCORE"

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 8]
```diff
@@ -140,9 +183,10 @@ def plot_ROC_curve(
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py:193]: Please remove "X_SCORE_mixed" if it is not used.

File: ugvc/mrd/srsnv_plotting_utils.py (hunk 6 above)
Comment (kenissur) [reply ugvc/mrd/srsnv_plotting_utils.py]: Actually, why are you removing this part?

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 9]
```diff
@@ -244,12 +288,18 @@ def LoD_training_summary(
...
         _,
         _,
-    ) = read_effective_coverage_from_sorter_json(cram_stats_file)
+    ) = read_effective_coverage_from_sorter_json(sorter_json_stats_file)
+
+    read_filter_correction_factor = 1
+    if ("X_FILTERED_COUNT" in df_tp) and ("X_READ_COUNT" in df_tp):
+        read_filter_correction_factor = (df_tp["X_FILTERED_COUNT"]).sum() / df_tp["X_READ_COUNT"].sum()
+    else:
+        logger.warning("X_FILTERED_COUNT or X_READ_COUNT no in dataset, read_filter_correction_factor = 1 in LoD")
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py]: Please use the enumeration from featuremap_utils instead of the "X_" strings

File: ugvc/mrd/srsnv_plotting_utils.py [hunk 10]
```diff
@@ -196,33 +240,33 @@ def LoD_training_summary(
...
-    Returns
-    -------
-    list
-        list of filters applied for LoD estimation
     """
 
     # apply filters to data
-    with open(filters_file, encoding="utf-8") as f:
-        filters = json.load(f)
+    if not ("is_mixed" in df):
```
Comment (kenissur) [inline ugvc/mrd/srsnv_plotting_utils.py:275]: is_mixed is deprecated, please adapt to the new formats

**Review by kenissur — COMMENTED** [review]
I started reviewing but then realized it's not rebased to mrd_task_force_tmp_main, could you please rebase once we converge this branch?

File: ugvc/mrd/srsnv_plotting_utils.py (hunk 2 above)
Comment (rinashalibo) [reply ugvc/mrd/srsnv_plotting_utils.py]: done

File: ugvc/mrd/srsnv_plotting_utils.py (hunk 3 above)
Comment (rinashalibo) [reply ugvc/mrd/srsnv_plotting_utils.py]: done

File: ugvc/mrd/srsnv_plotting_utils.py (hunk 5 above)
Comment (rinashalibo) [reply ugvc/mrd/srsnv_plotting_utils.py:156]: done

File: ugvc/mrd/srsnv_plotting_utils.py (hunk 9 above)
Comment (rinashalibo) [reply ugvc/mrd/srsnv_plotting_utils.py]: done
